tsl/src/compression/compression.c

```diff
@@ -1424,7 +1424,11 @@ tsl_compressed_data_in(PG_FUNCTION_ARGS)
 
 	decoded_len = pg_b64_dec_len(input_len);
 	decoded = palloc(decoded_len + 1);
-	decoded_len = pg_b64_decode(input, input_len, decoded);
+	decoded_len = pg_b64_decode_compat(input, input_len, decoded, decoded_len);
+
+	if (decoded_len < 0)
+		elog(ERROR, "could not decode data");
```

Comment:
If this error is thrown, it gives you very little context of what the issue is. I suggest clarifying:

```suggestion
		elog(ERROR, "could not decode base64-encoded compressed data");
```